Products/ZAlchemyConnector/SQLTemplate.py

```diff
@@ -0,0 +1,223 @@
+                if self.args[arg].get('value', None):
+                    return True
+                else:
+                    return False
+
+        # If not found in args list, verifies if it's a condition
+        # in a referenced attribute in class escope.
+        if hasattr(self, arg):
+            return self._check_condition(arg)
+
+        return False
+
+    def render(self, arg):
+        """Verify arg to render."""
+        if self.args:
+            if self.args.get(arg) and arg in list(self.args.keys()):
+                _value = self.args[arg].get('value', None)
+                if _value:
+                    return """{}""".format(_value)
+                else:
+                    return """"""
+
+        # If not found in args list, verifies if it's a condition
+        # in a referenced attribute in class' escope.
+        if hasattr(self, arg):
+            _cond = self._check_condition(arg)
+            if type(_cond) == bool:
+                return """{}""".format(int(_cond))
+
+            return """{}""".format(_cond)
+
+        return """"""
+
+    def set_condition(self, id, instance):
+        """Set condition in class' escope."""
+        if not hasattr(self, id):
+            setattr(self, id, instance)
+        else:
+            raise Exception('condition already exists.')
+
+    def _check_condition(self, id):
+        """Return condition."""
+        if hasattr(self, id):
+            cond = getattr(self, id)
+            if callable(cond):
+                return cond()
+            return cond
+
+        return False
+
+    def get_macro(self, id):
+        """Return macro if exists."""
+        if self.id_macros:
+            self._sql_macros = PageTemplateFile(
+                self.path + self.id_macros, globals())
+            return self._sql_macros.macros[id]
+
+        return
+
+    def count(self):
+        """Verify if query should group by count."""
+        if self.count:
+            return True
+        return False
+
+    def not_count(self):
+        """Verify if query should NOT group by count."""
+        if not self.count:
+            return True
+        return False
+
+    def render_template(self):
+        """Render PageTemplate."""
+        # re.sub(r'--\s+.*', '', file)
+
+        self._page_template = PageTemplateFile(self.path + self.id, globals())
+        sql_rendered = self._page_template()
+        sql_rendered = sql_rendered.strip()
+
+        if self.is_debug():
+            print(sql_rendered)
+
+        return sql_rendered
+
+    def query(self):
+        """Execute Query."""
+        if not self._get_wrapper:
+            raise Exception('Wrapper not configured.')
+
+        _query = Query(
+            id=self.id,
+            get_wrapper=self._get_wrapper,
+            template=self.render_template(),
+            arguments=self.get_args_as_to_query()
+        )
+
+        return _query(**self.get_args_as_to_exec()).dictionaries()
```

Comment:
Instead of writing this part like this, you could write it inside the ``__call__`` function.
```Python
def __call__(self):
    """Execute Query."""
    if not self._get_wrapper:
        raise Exception('Wrapper not configured.')

    _query = Query(
        id=self.id,
        get_wrapper=self._get_wrapper,
        template=self.render_template(),
        arguments=self.get_args_as_to_query()
    )

    return _query(**self.get_args_as_to_exec())
```
Then, to run the query you could just use:
```Python
q = SQLTemplate(...)
# actually runs the query
q()
```

Another thing is, remove the ``dictionaries()`` after the query call. This way the user can decide if they'll work with a list of dictionaries or the results object.